launcher/control.py

```diff
@@ -276,27 +287,44 @@ def on_project_changed(self, index):
         # Establish a connection to the project database
         self.log("Connecting to %s" % name, level=INFO)
 
-        frame = self.current_frame()
         project = io.find_one({"type": "project"})
 
         assert project is not None, "This is a bug"
 
-        frame["config"] = project["config"]
-
         # Get available project actions and the application actions
         actions = api.discover(api.Action)
         apps = lib.get_apps(project)
         self._registered_actions[:] = actions + apps
 
-        silos = io.distinct("silo")
-        self._model.push([
-            dict({
-                "name": silo,
-                "icon": DEFAULTS["icon"]["silo"],
-            })
-            for silo in sorted(silos)
-        ])
+        db_assets = io.find({"type": "asset"})
+        # Backwadrs compatbility with silo
+        silos = db_assets.distinct("silo")
+        if silos and None in silos:
+            silos = None
+
+        if not silos:
+            assets = list()
+            for asset in db_assets.sort("name", 1):
+                # _not_set_ is for cases when visualParent is not used
+                vis_p = asset.get("data", {}).get("visualParent", "_not_set_")
+                if vis_p is None:
+                    assets.append(asset)
+                elif vis_p == "_not_set_":
+                    assets.append(asset)
+
+            self._model.push([dict({
+                "_id": asset["_id"],
+                "name": asset["name"],
+                "type": asset["type"],
+                "icon": DEFAULTS["icon"]["asset"]
+            }) for asset in assets])
+
+        else:
+            self._model.push([dict({
+                "name": silo, "icon": DEFAULTS["icon"]["silo"], "type": "silo"
+            }) for silo in sorted(silos)])
 
+        frame = project
```

Comment:
Why removing `frame = self.current_frame()`, `frame["config"] = project["config"]` and use entire project document as frame ?

Is this necessary ? If not, I would like to keep it :D